### PR TITLE
Cleanup model before train

### LINKED ISSUE #1725: Checkpoint error with the same modelname and different training parameters
piece of the first SQL program:

``` 
WITH
  model.n_classes = 3,
  model.hidden_units = [10, 20],
  train.num_workers=2,
  train.num_ps=2,
  train.save_checkpoints_steps=20,
  train.epoch=10,
  train.batch_size=4,
  train.verbose=2
...
```
piece of the second SQL program:
```
WITH
  model.n_classes = 3,
  model.hidden_units = [10, 20],
  train.batch_size=4
...
```
The error logs:
``` text
Caused by op u'save/RestoreV2_1', defined at:
  File "entry.py", line 107, in <module>
    pai_val_table="alifin_jtest_dev.eGoQTeUEsCzXALtq")
  File "/apsara/TempRoot/Odps/alifin_jtest_dev_2020011609105861gqqx7bgu2_7f11e3f2_54ed_4e14_95a7_f8ad141fd671_AlgoTask_0_0/tensorflow@e15p13429.ew9#0/tensorflow_jobs/sqlflow_submitter/tensorflow/train.py", line 232, in train
    metric_names)
  File "/apsara/TempRoot/Odps/alifin_jtest_dev_2020011609105861gqqx7bgu2_7f11e3f2_54ed_4e14_95a7_f8ad141fd671_AlgoTask_0_0/tensorflow@e15p13429.ew9#0/tensorflow_jobs/sqlflow_submitter/tensorflow/train.py", line 162, in estimator_train_and_save
    result = tf.estimator.train_and_evaluate(classifier, train_spec, eval_spec)
  File "/usr/lib/python2.7/site-packages/tensorflow/python/estimator/training.py", line 471, in train_and_evaluate
    return executor.run()
  File "/usr/lib/python2.7/site-packages/tensorflow/python/estimator/training.py", line 610, in run
    return self.run_local()
  File "/usr/lib/python2.7/site-packages/tensorflow/python/estimator/training.py", line 711, in run_local
    saving_listeners=saving_listeners)
  File "/usr/lib/python2.7/site-packages/tensorflow/python/estimator/estimator.py", line 384, in train
    loss = self._train_model(input_fn, hooks, saving_listeners)
  File "/usr/lib/python2.7/site-packages/tensorflow/python/estimator/estimator.py", line 1241, in _train_model
    return self._train_model_default(input_fn, hooks, saving_listeners)
  File "/usr/lib/python2.7/site-packages/tensorflow/python/estimator/estimator.py", line 1343, in _train_model_default
    saving_listeners)
  File "/usr/lib/python2.7/site-packages/tensorflow/python/estimator/estimator.py", line 1574, in _train_with_estimator_spec
    log_step_count_steps=log_step_count_steps) as mon_sess:
  File "/usr/lib/python2.7/site-packages/tensorflow/python/training/monitored_session.py", line 523, in MonitoredTrainingSession
    stop_grace_period_secs=stop_grace_period_secs)
  File "/usr/lib/python2.7/site-packages/tensorflow/python/training/monitored_session.py", line 950, in __init__
    stop_grace_period_secs=stop_grace_period_secs)
  File "/usr/lib/python2.7/site-packages/tensorflow/python/training/monitored_session.py", line 672, in __init__
    self._sess = _RecoverableSession(self._coordinated_creator)
  File "/usr/lib/python2.7/site-packages/tensorflow/python/training/monitored_session.py", line 1214, in __init__
    _WrappedSession.__init__(self, self._create_session())
  File "/usr/lib/python2.7/site-packages/tensorflow/python/training/monitored_session.py", line 1219, in _create_session
    return self._sess_creator.create_session()
  File "/usr/lib/python2.7/site-packages/tensorflow/python/training/monitored_session.py", line 829, in create_session
    self.tf_sess = self._session_creator.create_session()
  File "/usr/lib/python2.7/site-packages/tensorflow/python/training/monitored_session.py", line 576, in create_session
    self._scaffold.finalize()
  File "/usr/lib/python2.7/site-packages/tensorflow/python/training/monitored_session.py", line 226, in finalize
    self._saver.build()
  File "/usr/lib/python2.7/site-packages/tensorflow/python/training/saver.py", line 1235, in build
    self._build(self._filename, build_save=True, build_restore=True)
  File "/usr/lib/python2.7/site-packages/tensorflow/python/training/saver.py", line 1275, in _build
    build_save=build_save, build_restore=build_restore)
  File "/usr/lib/python2.7/site-packages/tensorflow/python/training/saver.py", line 899, in _build_internal
    restore_sequentially, reshape)
  File "/usr/lib/python2.7/site-packages/tensorflow/python/training/saver.py", line 557, in _AddShardedRestoreOps
    name="restore_shard"))
  File "/usr/lib/python2.7/site-packages/tensorflow/python/training/saver.py", line 504, in _AddRestoreOps
    restore_sequentially)
  File "/usr/lib/python2.7/site-packages/tensorflow/python/training/saver.py", line 980, in bulk_restore
    return io_ops.restore_v2(filename_tensor, names, slices, dtypes)
  File "/usr/lib/python2.7/site-packages/tensorflow/python/ops/gen_io_ops.py", line 1707, in restore_v2
    shape_and_slices=shape_and_slices, dtypes=dtypes, name=name)
  File "/usr/lib/python2.7/site-packages/tensorflow/python/framework/op_def_library.py", line 787, in _apply_op_helper
    op_def=op_def)
  File "/usr/lib/python2.7/site-packages/tensorflow/python/util/deprecation.py", line 488, in new_func
    return func(*args, **kwargs)
  File "/usr/lib/python2.7/site-packages/tensorflow/python/framework/ops.py", line 3300, in create_op
    op_def=op_def)
  File "/usr/lib/python2.7/site-packages/tensorflow/python/framework/ops.py", line 1770, in __init__
    self._traceback = tf_stack.extract_stack()
InvalidArgumentError (see above for traceback): Restoring from checkpoint failed. This is most likely due to a mismatch between the current graph and the graph from the checkpoint. Please ensure that you have not altered the graph expected based on the checkpoint. Original error:

tensor_name = dnn/hiddenlayer_1/bias; shape in shape_and_slice spec [10] does not match the shape stored in checkpoint: [20]
	 [[node save/RestoreV2_1 (defined at /apsara/TempRoot/Odps/alifin_jtest_dev_2020011609105861gqqx7bgu2_7f11e3f2_54ed_4e14_95a7_f8ad141fd671_AlgoTask_0_0/tensorflow@e15p13429.ew9#0/tensorflow_jobs/sqlflow_submitter/tensorflow/train.py:162)  = RestoreV2[dtypes=[DT_FLOAT, DT_FLOAT, DT_FLOAT, DT_FLOAT, DT_FLOAT, DT_FLOAT, DT_FLOAT, DT_FLOAT, DT_FLOAT, DT_FLOAT, DT_FLOAT, DT_FLOAT], _device="/job:localhost/replica:0/task:0/device:CPU:0"](_arg_save/Const_0_0, save/RestoreV2_1/tensor_names, save/RestoreV2_1/shape_and_slices)]]
```

### PR DESCRIPTION
**Pull request by Yancey0623**
fixed #1725

### CODE REVIEW

File: pkg/sql/alisa_submitter.go [hunk 1]
```diff
@@ -112,20 +118,35 @@ func (s *alisaSubmitter) submit(program, alisaCode string) error {
...
 	if err != nil {
 		return err
 	}
 	if e := bucket.PutObjectFromFile(resourceName, filepath.Join(s.Cwd, tarball)); e != nil {
 		return err
 	}
-	defer func() {
-		bucket.DeleteObject(resourceName)
-	}()
+	defer bucket.DeleteObject(resourceName)
```
Comment (typhoonzero) [inline pkg/sql/alisa_submitter.go:129]: maybe change `cleanUpModel` to `deleteOSSObject(path)` and call it here too?

**Review by typhoonzero — APPROVED** [review]
LGTM with one enhancement advice

File: pkg/sql/alisa_submitter.go (hunk 1 above)
Comment (Yancey0623) [reply pkg/sql/alisa_submitter.go:129]: There are two OSS bucket configurations here: **Checkpoint Dir** which used in PAI-TF and **Alisa Resource Bucket** which used to store Alisa resource.

Maybe a clearer way is to initialize two `OSSClient` fields in `AlisaSubmitter` to operate OSS objects?

Comment (typhoonzero) [reply pkg/sql/alisa_submitter.go:129]: Maybe use `deleteOSSObject(bucket, path)` should work?

**Pull request merged**